### PR TITLE
fixes #66

### LINKED ISSUE #66: Module Fixes: sample
There was some slight rot on the `sample` module.  There was a bug using `--infile`.  And it was still using the tuple formation to access amount.

**AC**

- [x] Enable `--infile` to work correctly so you don't have to pipe into `sample`
- [x] Fix `balance[0]` to become `balance["amount"]`

### PR DESCRIPTION
**Pull request by inertia186**
Closes #66

### CODE REVIEW

File: tinman/sample.py [hunk 1]
```diff
@@ -32,7 +32,7 @@ def main(argv):
         outfile = sys.stdout
     else:
         outfile = open(args.outfile, "w")
-    json.dump(snapshot, f, separators=(",", ":"))
+    json.dump(snapshot, outfile, separators=(",", ":"))
```
Comment (relativityboy) [inline tinman/sample.py]: `outfile` might be undefined here.

Comment (inertia186) [reply tinman/sample.py]: I *think* `outfile` has definite-assignment.

Comment (relativityboy) [reply tinman/sample.py]: OIC. The horror of that code confused my brain...

```
if args.outfile == "-":
        outfile = sys.stdout
    else:
        outfile = open(args.outfile, "w")
```
Any idea why not
```
outfile = sys.stdout
if args.outfile != "-":
  outfile = open(args.outfile, "w")        
```

Comment (inertia186) [reply tinman/sample.py]: Because there's no need to worry about scope.

Comment (relativityboy) [reply tinman/sample.py]: Entering into a block when there's no advantage seems wasteful. Extra code, extra logic.

**Pull request merged**